feature-libs/cart/quick-order/components/cart-quick-form/cart-quick-form.component.ts

```diff
@@ -0,0 +1,149 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  OnDestroy,
+  OnInit,
+} from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import {
+  ActiveCartService,
+  Cart,
+  CartAddEntryFailEvent,
+  CartAddEntrySuccessEvent,
+  EventService,
+  GlobalMessageService,
+  GlobalMessageType,
+} from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Component({
+  selector: 'cx-cart-quick-order-form',
+  templateUrl: './cart-quick-form.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CartQuickOrderFormComponent implements OnInit, OnDestroy {
+  quickOrderForm: FormGroup;
+  cartIsLoading$: Observable<boolean> = this.activeCartService
+    .isStable()
+    .pipe(map((loaded) => !loaded));
+  cart$: Observable<Cart> = this.activeCartService.getActive();
+  min = 1;
+
+  private subscription: Subscription = new Subscription();
+  private minQuantityValue: number = 1;
+
+  constructor(
+    protected activeCartService: ActiveCartService,
+    protected eventService: EventService,
+    protected formBuilder: FormBuilder,
+    protected globalMessageService: GlobalMessageService
+  ) {}
+
+  ngOnInit(): void {
+    this.buildForm();
+    this.watchQuantityChange();
+    this.watchAddEntrySuccessEvent();
+    this.watchAddEntryFailEvent();
```

Comment:
Can we do the same thing we did with addToCart and entries that we only listen to those events when we submit form and we unsubscribe once it completes?